src/tasks/engine.ts

```diff
@@ -31,22 +59,59 @@ export class SafeGarboEngine extends BaseGarboEngine {
   }
 }
 
-export function runSafeGarboTasks(tasks: Task[]): void {
-  const engine = new SafeGarboEngine(tasks);
+export class BarfTaskEngine extends BaseGarboEngine<never, BarfTask> {
+  failures = 0;
 
-  try {
-    engine.run();
-  } finally {
-    engine.destruct();
+  available(task: BarfTask): boolean {
+    const validSobrieties = [Sobriety.EITHER, sober() ? Sobriety.SOBER : Sobriety.DRUNK];
+    return validSobrieties.includes(task.sobriety) && (task.ready === undefined || task.ready());
```

Comment:
```suggestion
    return validSobrieties.includes(task.sobriety) && (task.ready?.() ?? true);
```
Maybe this is dumb instead of being cool. But I think it's cool.